presto-parser/src/main/java/com/facebook/presto/sql/tree/TemplateExpressionRewriter.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+public class TemplateExpressionRewriter
+{
+    private static int counter;
```

Comment:
Since we are referencing counter from the `rewrite` method, counter needs to be static to be referenced from a static context.

Reply:
Ok but static means it belongs to the class, I think here it should belong to the visitor. Because otherwise I am resetting in one class and incrementing in another class.